query/src/main/java/org/infinispan/query/dsl/embedded/impl/BaseIckleFilterIndexingServiceProvider.java

```diff
@@ -251,6 +251,9 @@ public void onFilterResult(Object userContext, Object eventType, Object instance
          }
 
          boolean conversionDone = false;
+         if (invocations == null) {
```

Comment:
@anistor I was getting NPEs on the loop below when running the HotRodCacheContinuousQueries test on the server, so I added this. It fixes the issue and doesn't seem to cause side-effects.
The exception
`
java.lang.NullPointerException: null
	at org.infinispan.query.dsl.embedded.impl.BaseIckleFilterIndexingServiceProvider$Callback.onFilterResult(BaseIckleFilterIndexingServiceProvider.java:254) ~[infinispan-query-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.objectfilter.impl.predicateindex.MatcherEvalContext.notifyDeltaSubscribers(MatcherEvalContext.java:182) ~[infinispan-objectfilter-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.objectfilter.impl.BaseMatcher.matchDelta(BaseMatcher.java:116) ~[infinispan-objectfilter-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.query.continuous.impl.IckleContinuousQueryFilterIndexingServiceProvider.matchEvent(IckleContinuousQueryFilterIndexingServiceProvider.java:85) ~[infinispan-query-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.query.dsl.embedded.impl.BaseIckleFilterIndexingServiceProvider$FilteringListenerInvocation.invoke(BaseIckleFilterIndexingServiceProvider.java:319) ~[infinispan-query-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.notifications.cachelistener.CacheNotifierImpl.doNotifyVisited(CacheNotifierImpl.java:572) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.notifications.cachelistener.CacheNotifierImpl.notifyCacheEntryVisited(CacheNotifierImpl.java:559) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.EntryWrappingInterceptor.lambda$new$1(EntryWrappingInterceptor.java:139) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.BaseAsyncInterceptor.invokeNextThenApply(BaseAsyncInterceptor.java:86) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.EntryWrappingInterceptor.visitDataReadCommand(EntryWrappingInterceptor.java:217) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.EntryWrappingInterceptor.visitGetCacheEntryCommand(EntryWrappingInterceptor.java:211) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.commands.read.GetCacheEntryCommand.acceptVisitor(GetCacheEntryCommand.java:34) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.BaseAsyncInterceptor.invokeNext(BaseAsyncInterceptor.java:59) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.locking.NonTransactionalLockingInterceptor.visitDataReadCommand(NonTransactionalLockingInterceptor.java:34) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.locking.AbstractLockingInterceptor.visitGetCacheEntryCommand(AbstractLockingInterceptor.java:112) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.commands.read.GetCacheEntryCommand.acceptVisitor(GetCacheEntryCommand.java:34) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.BaseAsyncInterceptor.invokeNextAndHandle(BaseAsyncInterceptor.java:188) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.BaseStateTransferInterceptor.handleReadCommand(BaseStateTransferInterceptor.java:193) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.BaseStateTransferInterceptor.visitGetCacheEntryCommand(BaseStateTransferInterceptor.java:174) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.commands.read.GetCacheEntryCommand.acceptVisitor(GetCacheEntryCommand.java:34) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.BaseAsyncInterceptor.invokeNext(BaseAsyncInterceptor.java:59) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.CacheMgmtInterceptor.visitDataReadCommand(CacheMgmtInterceptor.java:115) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.CacheMgmtInterceptor.visitGetCacheEntryCommand(CacheMgmtInterceptor.java:98) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.commands.read.GetCacheEntryCommand.acceptVisitor(GetCacheEntryCommand.java:34) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.BaseAsyncInterceptor.invokeNext(BaseAsyncInterceptor.java:59) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.DDAsyncInterceptor.handleDefault(DDAsyncInterceptor.java:53) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.DDAsyncInterceptor.visitGetCacheEntryCommand(DDAsyncInterceptor.java:111) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.commands.read.GetCacheEntryCommand.acceptVisitor(GetCacheEntryCommand.java:34) ~[infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.BaseAsyncInterceptor.invokeNextAndExceptionally(BaseAsyncInterceptor.java:128) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.InvocationContextInterceptor.visitCommand(InvocationContextInterceptor.java:89) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.BaseAsyncInterceptor.invokeNext(BaseAsyncInterceptor.java:61) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.DDAsyncInterceptor.handleDefault(DDAsyncInterceptor.java:53) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.DDAsyncInterceptor.visitGetCacheEntryCommand(DDAsyncInterceptor.java:111) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.commands.read.GetCacheEntryCommand.acceptVisitor(GetCacheEntryCommand.java:34) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.DDAsyncInterceptor.visitCommand(DDAsyncInterceptor.java:49) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.interceptors.impl.AsyncInterceptorChainImpl.invokeAsync(AsyncInterceptorChainImpl.java:226) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.cache.impl.CacheImpl.getCacheEntryAsync(CacheImpl.java:548) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.cache.impl.CacheImpl.getCacheEntryAsync(CacheImpl.java:541) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.cache.impl.AbstractDelegatingAdvancedCache.getCacheEntryAsync(AbstractDelegatingAdvancedCache.java:283) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.cache.impl.EncoderCache.getCacheEntryAsync(EncoderCache.java:383) [infinispan-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.server.hotrod.CacheRequestProcessor.getInternal(CacheRequestProcessor.java:74) [infinispan-server-hotrod-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.server.hotrod.CacheRequestProcessor.get(CacheRequestProcessor.java:69) [infinispan-server-hotrod-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.server.hotrod.HotRodDecoder.switch1(HotRodDecoder.java:766) [infinispan-server-hotrod-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.server.hotrod.HotRodDecoder.switch1_0(HotRodDecoder.java:154) [infinispan-server-hotrod-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at org.infinispan.server.hotrod.HotRodDecoder.decode(HotRodDecoder.java:143) [infinispan-server-hotrod-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:505) [netty-codec-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:444) [netty-codec-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:283) [netty-codec-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at org.infinispan.server.core.transport.StatsChannelHandler.channelRead(StatsChannelHandler.java:26) [infinispan-server-core-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1421) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930) [netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:794) [netty-transport-native-epoll-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:424) [netty-transport-native-epoll-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:326) [netty-transport-native-epoll-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.39.Final.jar:4.1.39.Final]
	at java.lang.Thread.run(Thread.java:834) [?:?]
`